### PR TITLE
Types for Passport CustomStrategy

### PR DESCRIPTION
**Pull request by zarly**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -23,5 +23,11 @@
     "devDependencies": {
         "dtslint": "github:Microsoft/dtslint#production",
         "types-publisher": "Microsoft/types-publisher#production"
+    },
+    "dependencies": {
+        "express": "^4.16.4",
```
[the remark this answers is not in the document]
Comment (zarly) [reply package.json]: done

File: types/passport-custom/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Type definitions for passport-custom 1.0.5
```
[the remark this answers is not in the document]
Comment (zarly) [reply types/passport-custom/index.d.ts]: done

File: types/passport-custom/tslint.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        "semicolon": false,
+        "space-before-function-paren": false,
+        "space-within-parens": false,
+        "strict-export-declare-modifiers": false,
+        "trim-file": false,
+        "triple-equals": false,
+        "typedef-whitespace": false,
+        "unified-signatures": false,
+        "void-return": false,
+        "whitespace": false
```
[the remark this answers is not in the document]
Comment (zarly) [reply types/passport-custom/tslint.json]: done

**Comment by zarly** [comment]
fixed

**Comment by DanielRosenwasser** [comment]
Please fix up the lint errors that are being reported.

Additionally, just as a heads up, your commits don't seem to be associated with your GitHub account. While this isn't technically a problem, you might care if you want more appropriate attribution. You can either make sure your GitHub account is associated with the email address you're using for your commits, or rebase and amend your commits to fix the author name and email.

**Pull request closed without merging**